ee/clickhouse/queries/paths/paths.py

```diff
@@ -94,18 +88,49 @@ def get_path_query_by_funnel(self, funnel_filter: Filter):
         {path_query}
         """
 
-    def get_start_point_filter(self) -> Tuple[str, Dict]:
-
-        if not self._filter.start_point:
-            return "", {"target_point": None}
-
-        return "WHERE arrayElement(limited_path, 1) = %(target_point)s", {"target_point": self._filter.start_point}
-
-    def get_end_point_filter(self) -> Tuple[str, Dict]:
-        if not self._filter.end_point:
-            return "", {"target_point": None}
-
-        return "WHERE arrayElement(limited_path, -1) = %(target_point)s", {"target_point": self._filter.end_point}
+    def get_target_point_filter(self) -> Tuple[str, Dict]:
+        conditions = []
+        params: Dict[str, Union[str, None]] = {"target_point": None}
+
+        if self._filter.end_point and self._filter.start_point:
+            conditions.append("arrayElement(limited_path, -1) = %(target_point)s")
+            params.update({"target_point": self._filter.end_point})
+
+            conditions.append("arrayElement(limited_path, 1) = %(secondary_target_point)s")
+            params.update({"secondary_target_point": self._filter.start_point})
+        elif self._filter.end_point:
+            conditions.append("arrayElement(limited_path, -1) = %(target_point)s")
+            params.update({"target_point": self._filter.end_point})
+        elif self._filter.start_point:
+            conditions.append("arrayElement(limited_path, 1) = %(target_point)s")
+            params.update({"target_point": self._filter.start_point})
+
+        if not conditions:
+            return "", params
+
+        return f"WHERE {' AND '.join(conditions)}", params
+
+    def get_target_clause(self) -> str:
+        if self._filter.end_point and self._filter.start_point:
+            return """
+            , indexOf(compact_path, %(secondary_target_point)s) as start_target_index
```

Comment:
This makes sense to me.

An alternative here might be to always keep these in SQL, and just update params for `target_point` and `secondary_target_point` ? Seems easier to read & use, at a very minimal cost of time-complexity. Thoughts?

Reply:
wouldn't need the first half of the clause if we only have one target though

Reply:
Yeah, the idea was that even if we don't need it, the "start index" value would be -1, which means the if conditional would disregard it.

So we wouldn't need to do any modifications during query build time, and all things would be taken care of during execution time.

It's a minor point though, I'm okay with either right now, we'll find out more once we add more features here and see how this fares.